unittest/gunit/thread_utils.h

```diff
@@ -115,4 +115,22 @@ class Notification
 
 }  // namespace thread
 
+extern void disable_core_dumps(void);
+extern void restore_core_dump_rlimit(void);
+
+
+#define MY_EXPECT_DEATH_IF_SUPPORTED(statement, regex)  \
+  do {                                                  \
+    disable_core_dumps();                               \
```

Comment:
Implemented. No ASSERT_EQ in set/getrlimit failure handling, as these macros then make the constructor/destructor to return a value